binary_search_tree/tree.py

```diff
@@ -1,59 +1,119 @@
 class TreeNode:
-    def __init__(self, key, val = None):
+    def __init__(self, key, val=None):
         if val == None:
             val = key
 
         self.key = key
         self.value = val
         self.left = None
         self.right = None
-        
 
 
 class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
-    def add(self, key, value = None):
-        pass
+    # Time Complexity: O(log n)
+    # Space Complexity: O(1)
+    def add(self, key, value=None):
```

Comment:
👍  But the space complexity is O(log n) for a balanced tree due to the call stack because of the recursive nature of this solution.  For an unbalanced tree this would be O(n) space complexity.